### ISSUE TITLE
Hard coded user agent string increases detection

### ISSUE THREAD

**Issue by pyrohaz4good** [Veil-Framework/Veil-Evasion#116]
I'm a huge fan of veil and use it on pentests and in red teaming. However, I'm now seeing my clients take the user agent strings here and setup detection alerts. 

I'm thinking of forking and adding a new powershell module with the ability to detect which version of IE is installed (everyone has IE that has powershell) and set the user agent that way. The disadvantage of course is the payload size of storing the lookup for all those strings. 

If you guys get to it first, it would be a great feature to dynamically set the user agent based upon enumerating which browsers are installed. Maybe its beyond the scope of what you want to do with veil but it would be cool.

**Comment by pyrohaz4good** [comment]
I thought through this a bit more and realized figuring out the user agent string based on the system is tricky if not impossible. What is possible however is to on the victim machine, before using system.net.webclient, launch an IE com object as invisible and navigate to a listening metasploit HTTP server that determines the UA then serves back the proper powershell payload patched with the user agent. Powershell can read the encoded powershell stager from the body property of the IE document object then launch a new powershell process to execute it (which then steps through the normal rev_http steps). Only change for the veil payload would be making the UA a parameter. On msf however, I'd have to write a remote browser exploit module for it to work.

Thus all communication would be done with the UA that has been seen before.

**Comment by ChrisTruncer** [comment]
It's an option, definitely.  As of now, I'm a little overloaded where I probably won't be able to take this on currently, but we'd always accept a pull request if you do make a payload module that basically carries this out.  The problem is, you'd have to configure stuff server side in order for this to work too.  Now, we have the staged powershell payload, so we're not opposed to that either.

**Comment by pyrohaz4good** [comment]
Actually I had a few hours this weekend and just finished up creating a new module. Seems to work well in my limited tests with Windows Server 2012. I'll do a pull request.

I also did the server side part of this in MSF so I'll need to do a pull request for them as well. On the MSF side, I just modified the standard reverse_http and added an option to send a new type of request. Maybe I can send my branch name once I check it in so you guys can test?
